algorithms/src/snark/marlin/ahp/verifier/messages.rs

```diff
@@ -39,18 +46,18 @@ pub struct SecondMessage<F> {
 }
 
 /// Third message of the verifier.
-#[derive(Copy, Clone, Debug)]
+#[derive(Clone, Debug)]
 pub struct ThirdMessage<F> {
-    /// Randomizer for the h-polynomial for `B`.
-    pub r_b: F,
-    /// Randomizer for the h-polynomial for `C`.
-    pub r_c: F,
+    /// Randomizers for the h-polynomial for `A_i`, `B_i`, `C_i` for circuit i.
+    pub r_a: Vec<F>,
```

Comment:
I think we can avoid the `r_a`'s, no?

Reply:
As far as I understand we can only avoid the *first* r_a_0 (in the protocol-docs this would be `\deta_a_0`). At which point we might as well keep that value (`F::one`) around as well. 

I'm summing all `g_M`'s using these randomizers

Reply:
Oh you're totally right, I misread and thought that this was for the first round (i.e. the $\eta$'s)